packages/server-api/src/index.ts

```diff
@@ -172,6 +172,17 @@ export interface ServerAPI extends PluginServerApp {
     ) => void
   }) => void
   getSerialPorts: () => Promise<Ports>
+
+  getFeatures: () => {
+    apis: string[]
+    plugins: {
+      id: string
+      name: string
+      version: string
+    }
+  }
+  registerFeature: (type: string, id: string) => void
```

Comment:
Features return type fixes the response to contain exactly apis and plugins and you can only register apis, so what do we need the type parameter for? what use cases do you envision it for? I am not fond of the arbitrary string types.

Registerfeature is now a separate function, but we have server apis for registering things like resource handlers and apis. Could we implement this so that features are reported based on registering the services, as a side effect and not something separate you need to call? i fear adoption, if this is extra and not part of what feature providers already do.

Reply:
> Features return type fixes the response to contain exactly apis and plugins and you can only register apis, so what do we need the type parameter for? what use cases do you envision it for? I am not fond of the arbitrary string types.

It's there only because there maybe other "features" in future that are not APIs or plugins (e.g. maybe operations like trip logging).
I don't have any in mind just a placeholder for possible future ideas.

Reply:
> Registerfeature is now a separate function, but we have server apis for registering things like resource handlers and apis. Could we implement this so that features are reported based on registering the services, as a side effect and not something separate you need to call? i fear adoption, if this is extra and not part of what feature providers already do.

Do you mean use something like setting an attribute instead to flag that a plugin is a provider for a feature?

Reply:
Removed the `registerFeature()` method.
Plugins implementing an API can set the `feature` attribute and it will be detected.